include/benchmark/benchmark.h

```diff
@@ -1181,6 +1201,39 @@ class CSVReporter : public BenchmarkReporter {
   std::set< std::string > user_counter_names_;
 };
 
+class HTMLReporter : public BenchmarkReporter {
```

Comment:
i'm still a little uncomfortable with this as a reporter. given we output in JSON (and can add more fields there if necessary) an HTML view could be a server that runs as a standalone binary pointed at a JSON result. That could then be generalised into a service to which people might upload their results and see them visualised over time.